indico/modules/rb/tasks.py

```diff
@@ -111,9 +62,17 @@ def roombooking_occurrences():
 
     try:
         for occ in occurrences:
-            notify_upcoming_occurrence(occ)
-            occ.notification_sent = True
-            if occ.reservation.repeat_frequency == RepeatFrequency.DAY:
-                occ.reservation.occurrences.update({'notification_sent': True})
+            same_user_occurrences = []
+            for other in occurrences:
```

Comment:
Well, my biggest complaint is not about query performance, it's about the way in which they're performed (cartesian product of all occurrences). We can probably use a GROUP BY query. And even if we do it client-side, we can use a dictionary to index the occurrences by user.